### ISSUE TITLE
SQL generated for datetime "is empty" filter on MSSQL is not correct

### ISSUE THREAD

**Issue by notrom** [metabase/metabase#41158]
### Describe the bug

I upgraded to v0.49.3 and have found a significant issue when using MS SQL Server as a data source.

When setting a question filter on a datetime field to "is empty", Metabase is generating incorrect SQL:
![image](https://github.com/metabase/metabase/assets/4504437/e57268fd-51c6-4708-93ed-6483c3f8395c)
Uses this SQL:
![image](https://github.com/metabase/metabase/assets/4504437/8522edb9-b9cf-4375-b91d-b5500eea76d8)

The where clause is:
```sql
WHERE
    "xxx"."yyy"."IssueDate" = CAST(NULL AS datetime)
```

Equality comparisons should never be used for for NULL, is needs to be `"xxx"."yyy"."IssueDate" = is NULL`.

This is a blocking issue for us, I rolled back as soon as this was found.

### To Reproduce

1. Create a new question on a MSSQL data source table with a datetime field
2. Filter for any datetime fields "is empty" (see note in additional context)
3. Visualise, no data will be shown, that's not expected in my case.
4. View the SQL and the, see that it's using `"xxx"."yyy"."zzzDate" = CAST(NULL AS datetime)` for the empty check
5. Convert to SQL, change where clause to  `"xxx"."yyy"."IssueDate" = is NULL`
6. Visualise, data is returned as expected.

### Expected behavior

Correct SQL will be generated, `dateField is NULL` is the way to check for NULLs.

### Logs

Nothing

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "en-NZ",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "Cp1252",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "17.0.2+8",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "17.0.2",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "17.0.2+8",
    "os.name": "Windows 10",
    "os.version": "10.0",
    "user.language": "en",
    "user.timezone": "Pacific/Auckland"
  },
  "metabase-info": {
    "databases": [
      "h2",
      "postgres",
      "sqlserver"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "12.1"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.7.2"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2024-04-02",
      "tag": "v0.49.3",
      "hash": "dba0992"
    },
    "settings": {
      "report-timezone": "Pacific/Auckland"
    }
  }
}
```

### Severity

Major, we can't use a system that can't find rows with a missing date.

### Additional context

Sample dataset queries are generating the correct SQL so it seems to be MS SQL server specific. I'd guess related to the changes recently to date filtering applying timezones where it wasn't appropriate.

Setting a filter for a date "is empty" is harder than it should be. I do these steps:
- filter date field
- Exclude
- Not Empty
![image](https://github.com/metabase/metabase/assets/4504437/9c3a6b38-eb02-4e26-9c5e-2a04265c8e7d)

To me that's cumbersome and non-intuitive

**Comment by perivamsi** [comment]
@notrom this will be fixed in 49.4 https://github.com/metabase/metabase/pull/40957

**Comment by perivamsi** [comment]
Closing as duplicate of https://github.com/metabase/metabase/issues/41005
